src/remote-config/remote-config-api-client.ts

```diff
@@ -125,6 +134,42 @@ export class RemoteConfigApiClient {
       });
   }
 
+  public validateTemplate(template: RemoteConfigContent): Promise<RemoteConfigContent> {
+    return this.getUrl()
+      .then((url) => {
+        const headers = FIREBASE_REMOTE_CONFIG_PUT_HEADERS;
+        headers["If-Match"] = template.etag;
```

Comment:
Mutating a constant

Reply:
Aren't we just mutating a property here? Not the constant itself?
As an alternative, we could create the headers in-place:
```
headers: {
'X-Firebase-Client': 'fire-admin-node/<XXX_SDK_VERSION_XXX>',
If-Match: etag, 
'Accept-Encoding': 'gzip',
}
```

Reply:
We could also decompose the objects (rest/spread) and use it as below. Any preferences?
```
headers: { ...FIREBASE_REMOTE_CONFIG_GET_HEADERS, 'If-Match': template.etag },
```

Reply:
Decomposition looks a little better IMO.